Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -144,16 +161,15 @@ void CreateWorkspace::exec() {
   }
 
   // Create the OutputWorkspace
+  auto storageMode = Parallel::fromString(getProperty("ParallelStorageMode"));
+  IndexInfo indexInfo(nSpec, storageMode, communicator());
```

Comment:
I guess an alternative is going through the property mechanism, however I think you run into the same design problems. For example, if you overloaded `declareProperty` within `Algorithm` to accept handle the special case of a WorkspaceProperty and a storage mode. Then you could create the indexInfo behind the scenes. 

Alternatively, you can specify a storage mode in the WorkspaceProperty itself.